core/src/main/java/org/infinispan/util/concurrent/ConditionFuture.java

```diff
@@ -100,17 +100,19 @@ public void updateAsync(T value, Executor executor) {
       try {
          executor.execute(() -> checkConditions(value));
       } catch (Throwable t) {
-         List<CompletableFuture<?>> completed;
-         synchronized (futures) {
-            completed = new ArrayList<>(futures.size());
-            for (Data data : futures.values()) {
-               data.cancelFuture.cancel(false);
-               completed.add(data);
-            }
-         }
-         for (CompletableFuture<?> future : completed) {
-            future.completeExceptionally(t);
-         }
+         completeAllExceptionally(t);
+      }
+   }
+
+   private void completeAllExceptionally(Throwable t) {
+      List<Data> completed;
+      synchronized (futures) {
+         completed = new ArrayList<>(futures.size());
```

Comment:
why not use `new ArrayList<>(futures.values());`

Reply:
Done